src/CapabilityService.WebApi/Features/Shared/Configuration/DependencyInjection.cs

```diff
@@ -0,0 +1,35 @@
+using DFDS.CapabilityService.WebApi.Features.Shared.Infrastructure.Messaging;
+using DFDS.CapabilityService.WebApi.Features.Shared.Infrastructure.Persistence;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace DFDS.CapabilityService.WebApi.Features.Shared.Configuration
+{
+    public static class DependencyInjection
+    {
+        public static void AddShared(
```

Comment:
In some cases it might be nice to have Feature specific options. Right now we cant control the lifetime of the DbContext that is added, which means it will always be scoped by default, however there could be cases where people want it to be Transient or Singleton which would force them to workaround this method and manually register things. 

e.g:

public static IServiceCollection AddMyProjData(
    this IServiceCollection services,
    Action<MyProjDataOptionsBuilder> optionsBuilder)
{
    var myProjDataOptionsBuilder = new MyProjDataOptionsBuilder();
    optionsBuilder(myProjDataOptionsBuilder);

    services.Configure(optionsBuilder);

    services.AddDbContext<MyProjDbContext>(contextOptions => contextOptions
        .UseLazyLoadingProxies()
        .UseMySql(myProjDataOptionsBuilder.ConnectionString)
    );

    return services;
}